server/bin/pbench-unpack-tarballs.sh

```diff
@@ -110,45 +112,52 @@ function gen_work_list() {
     # First we find all the ${linksrc} directories
     for linksrc_dir in $(find ${ARCHIVE}/ -maxdepth 2 -type d -name ${linksrc}); do
         # Find all the links in a given ${linksrc} directory that are links to
-        # actual files (bad links are not emitted!).
-        find -L ${linksrc_dir} -type f -name '*.tar.xz' ${lb_arg} ${ub_arg} -printf "%s %p\n" 2>/dev/null >> ${list}.unsorted
+        # actual files (bad links are not emitted!).  For now, if it's a
+        # duplicate name, just punt and avoid producing an error.
+        find -L ${linksrc_dir} -type f -name '*.tar.xz' ! -name 'DUPLICATE__NAME*' ${lb_arg} ${ub_arg} -printf "%CY-%Cm-%CdT%CT %s %p\n" 2>/dev/null >> ${list}.unsorted
         if [[ ${lowerbound} == 0 ]]; then
             # Find all the links in the same ${linksrc} directory that don't
             # link to anything so that we can count them as errors below.
-            find -L $linksrc_dir -type l -name '*.tar.xz' -printf "%s %p\n" 2>/dev/null >> ${list}.unsorted
+            find -L $linksrc_dir -type l -name '*.tar.xz' ! -name 'DUPLICATE__NAME*' -printf "%CY-%Cm-%CdT%CT %s %p\n" 2>/dev/null >> ${list}.unsorted
         fi
     done
-    sort -n ${list}.unsorted > ${list}
+    sort -k 1 -r ${list}.unsorted > ${list}
     rm -f ${list}.unsorted
+    # Pad by one minute, the default smallest cronjob interval.
+    let max_seconds=${SECONDS}+60
+    return $(( ${max_seconds} * 2 ))
```

Comment:
So if the loop and the sort takes 5 seconds, this routine will return 2*(5+60) = 130 seconds of work time. Is that what you want or did you mean to say 2*5+60 = 70 seconds? Also, shouldn't we scale the work time differently depending on the bucket? Large tarballs take roughly the same time to be found and sorted, but they take longer to be unpacked.

Reply:
If we sort by time, and always  work from newer to older, isn't there a danger of starving the older tarballs?

Reply:
As we discussed face-to-face, we want the unpack loop to re-evaluate the list of tar balls to unpack if `${max_seconds}` has passed since it started unpacking tar balls.  The goal is to have a gross approximation of unpacking the most recent tar balls as soon as possible.  

The shortest amount of time in the current method that we could notice and unpack a tar ball is 1 minute, the minimum cronjob interval.   But this code does not use 1 minute as the maximum so that for the smaller tar balls we can process more of the existing set before we find a new set.  For the larger ones it doesn't really matter, as there are fewer of them.

As for starvation, yes this can starve out older tar balls, but that happens today but in reverse: older tar balls can starve out newer ones for many hours or days in the right situation.  Solving starvation properly is a much larger change, and out of scope for this change.